code/drasil-example/swhs/lib/Drasil/SWHS/GenDefs.hs

```diff
@@ -109,10 +111,11 @@ rocTempDerivArbVol :: [Sentence]
 rocTempDerivArbVol = [S "We consider an arbitrary" +:+. phrase vol,
   atStartNP (the volHtGen), S "is assumed constant. Then", eqN 1, S "can be written as"]
 
-rocTempDerivConsFlx :: Sentence -> [ConceptInstance] -> [Sentence]
-rocTempDerivConsFlx s assumps = [S "Where", 
+rocTempDerivConsFlx :: (HasUID r, HasRefAddress r, HasShortName r) => r
+ -> Sentence -> [ConceptInstance] -> [Sentence]
+rocTempDerivConsFlx r s assumps = [S "Where", 
   foldlList Comma List (map ch [htFluxIn, htFluxOut, inSA, outSA]),
-  S "are explained in" +:+. refS rocTempSimp, s, S "Assuming", 
+  S "are explained in" +:+. refS r, s, S "Assuming",  -- FIXME: refS r is a hack to share the rocTempSimpRC instance without causing friction between nopcm and swhs
```

Comment:
This `rocTempDerivConsFlx` reference expected the variant of `rocTempSimp` belonging to SWHS, but it was accidentally referenced in nopcm. So, when I changed the UIDs for the `rocTempSimp` in nopcm, this reference broke since the `rocTempSimp` of SWHS wasn't registered in NoPCM's ChunkDB. In other words, this was a reference bug that happened to not cause problems because the UIDs lined up unintentionally.

Reply:
Ah, I see how all the above changes 'line up' to here. What I don't quite understand is what is really going on here. At the implementation level, I see what's going on. It's at the "meaning" level that I'm not so sure.  What is the reference `r` supposed to point to? What's the knowledge that's being used / expected / woven together here?